Creiwry/kitten_project
# Stripe checkout add transaction rollback

### Ticket Link
https://trello.com/c/yIMz78bQ

### User Story
https://trello.com/c/HdXHuENc
https://trello.com/c/QxsqhYFh

### What does this PR do?
In the case of a failed creation of order or order items, the stripe payment for the order gets refunded. A refund method has been added to the checkout controller and logging has been implemented so that the store owner has a way to access refund id's.
It also authenticate that the user is signed in before each action in the controller.

### Where should the reviewer start?
All the changes have been made in the checkout controller.

### How should this be manually tested?
make an order fail (eg, temporarily remove user_id from order.new method) and try to complete a stripe transaction. You should see a flash message that the order has been refunded and you should see the log of the refund id in the log/development.log file

### Are there any breaking changes?
no
